code/game/objects/structures/training_machine.dm

```diff
@@ -0,0 +1,310 @@
+#define MIN_RANGE 1
+#define MIN_SPEED 1
+#define MAX_RANGE 7
+#define MAX_SPEED 10
+#define HITS_TO_KILL 9
+#define MIN_ATTACK_DELAY 10
+#define MAX_ATTACK_DELAY 15
+
+/**
+  * Machine that runs around wildly so people can practice clickin on things
+  *
+  * Can have a mob buckled on or a obj/item/target attached. Uses timers to move, which is necessary because it's movespeed
+  * needs to both be variable and potentially faster than SSFastProcess's tick speed, so hey this is what we got. 
+  * As a note, there generally shouldn't be too many timers in existance at a time, so there will only ever be
+  * one or two of these on the station.
+  */
+/obj/structure/training_machine
+	name = "AURUMILL-Brand MkII. Personnel Training Machine"
+	desc = "Used for combat training simulations. Accepts standard training targets. A pair of buckling straps are attached."
+	icon = 'icons/obj/objects.dmi'
+	icon_state = "training_machine"
+	can_buckle = TRUE
+	buckle_lying = 0
+	max_integrity = 200
+	///Is the machine moving? Setting this to FALSE will automatically call stop_moving()
+	var/moving = FALSE
+	///The distance the machine is allowed to roam from its starting point
+	var/range = 1
+	///A random spot within range that the trainer is trying to reach
+	var/turf/target_position
+	///The turf the machine was on when it was activated
+	var/turf/starting_turf
+	///How fast the machine moves. Cannot be higher than MAX_SPEED
+	var/move_speed = 1
+	///Reference to a potentially attached object, either a target, trainer toolbox, or syndicate toolbox
+	var/obj/item/attached_item
+	///Time between attacks when emagged
+	var/attack_cooldown = 50
```

Comment:
```suggestion
```
This could be removed, with usage of the cooldown defines, since it won't be var-editable anyway, as it's getting reset on every attack.